docs/new-feature-tutorial.md

```diff
@@ -11,12 +11,44 @@ real-time push system; the
 [real-time push and events](events-system.html) documentation has a
 detailed explanation of how everything works.
 
-## General Process in brief
+## General Process
+
+### Files impacted
+
+This tutorial will walk through adding a new feature to a Realm (an
+organization in Zulip). The following files are involved in the process:
+
+**Backend**
+- `zerver/model.py`: Defines the database model.
+- `zerver/lib/actions.py`: Contains code for updating and interacting with the database.
+- `zerver/lib/events.py`: Involved in updating the state of the application after
+  an event (for example, an administrator changing the value of a realm setting).
+- `zerver/views/realm.py`: Receives request variables from JavaScript and calls
+  `actions.py` code to update the database.
+
+**Frontend**
+- `static/templates/settings/organization-permissions-admin.handlebars`: defines
+   the structure of the admin permissions page (checkboxes for each organization
+   permission setting).
+- `static/js/settings_org.js`: handles organization setting form submission.
+- `static/js/server_events_dispatch.js`: handles events coming from the server
+  (ex: pushing an organization change to other open browsers and updating
+  the application's state).
+
+**Backend Testing**
+- `zerver/tests/test_events.py`: tests sending events (for example, the
+   updating of a realm setting).
```

Comment:
I'd swap the order here, putting `test_realm.py` first, and then having this be "tests for possible race bugs in the zerver/lib/events.py implementation"